fix panic for unsatisfiable version constraint

If `version` is set to an unsatisfiable constraint (for example, a version is specified that does not exist), a panic is currently raised:

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/blendle/kubecrt/helm.GetAcceptableVersion(0xc420dd2130, 0xb, 0xc420dd2148, 0x8, 0x0, 0x11, 0xc420bc1460, 0x19)
        /private/tmp/kubecrt-20170513-15659-nlayje/src/github.com/blendle/kubecrt/helm/version.go:35 +0x114
github.com/blendle/kubecrt/chart.locateChartPath(0xc420dd2130, 0xb, 0xc420dd2148, 0x8, 0x1075985, 0x4, 0xc4206e3bd0, 0x101699f)
        /private/tmp/kubecrt-20170513-15659-nlayje/src/github.com/blendle/kubecrt/chart/parser.go:190 +0x589
github.com/blendle/kubecrt/chart.(*Chart).compile(0xc420bba500, 0x7fff5fbffa4d, 0x11, 0x7fff5fbffa2c, 0x19, 0xc420bba680, 0x3a, 0x1, 0x1, 0x1, ...)
        /private/tmp/kubecrt-20170513-15659-nlayje/src/github.com/blendle/kubecrt/chart/parser.go:72 +0x61
github.com/blendle/kubecrt/chart.(*Chart).ParseChart(0xc420bba500, 0x7fff5fbffa4d, 0x11, 0x7fff5fbffa2c, 0x19, 0x0, 0x0, 0x0, 0x0, 0x0)
        /private/tmp/kubecrt-20170513-15659-nlayje/src/github.com/blendle/kubecrt/chart/parser.go:61 +0x253
github.com/blendle/kubecrt/chartsconfig.(*ChartsConfiguration).ParseCharts(0xc420db34a0, 0x0, 0x0, 0xc420db34a0, 0x0, 0x0)
        /private/tmp/kubecrt-20170513-15659-nlayje/src/github.com/blendle/kubecrt/chartsconfig/parser.go:49 +0xa6
main.main()
        /private/tmp/kubecrt-20170513-15659-nlayje/src/github.com/blendle/kubecrt/main.go:72 +0x92d
```

We should instead inform the user that an acceptable version could not be found in the currently supplied repositories.

--- a/chart/parser.go
+++ b/chart/parser.go
@@ -187,7 +187,7 @@ func locateChartPath(name, version string) (string, error) {
 
 	version, err = helm.GetAcceptableVersion(name, version)
 	if err != nil {
-		return "", fmt.Errorf("Failed to find chart versions: %s", err)
+		return "", err
 	}
 
 	filename, _, err := dl.DownloadTo(name, version, filepath.Dir(crepo))

--- a/helm/version.go
+++ b/helm/version.go
@@ -23,6 +23,10 @@ func GetAcceptableVersion(name, constraint string) (string, error) {
 		return "", err
 	}
 
+	if len(res) == 0 {
+		return "", fmt.Errorf(`unable to find chart named "%s"`, name)
+	}
+
 	search.SortScore(res)
 
 	if constraint != "" {
@@ -58,7 +62,7 @@ func buildIndex() (*search.Index, error) {
 func applyConstraint(version string, res []*search.Result) ([]*search.Result, error) {
 	constraint, err := semver.NewConstraint(version)
 	if err != nil {
-		return res, fmt.Errorf("an invalid version/constraint format: %s", err)
+		return res, fmt.Errorf("invalid chart version/constraint format: %s", err)
 	}
 
 	data := res[:0]
@@ -67,6 +71,10 @@ func applyConstraint(version string, res []*search.Result) ([]*search.Result, er
 		if err != nil || constraint.Check(v) {
 			data = append(data, r)
 		}
+	}
+
+	if len(data) == 0 {
+		return data, fmt.Errorf("unable to fulfil chart version constraint %s", version)
 	}
 
 	return data, nil

